plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -348,6 +351,27 @@ public void renameSchema(ConnectorSession session, String schemaName, String new
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
     }
 
+    @Override
+    public OptionalInt getMaxColumnNameLength(ConnectorSession session)
+    {
+        if (maxColumnNameLength != null) {
+            // According to JavaDoc of DatabaseMetaData#getMaxColumnNameLength a value of 0 signifies that the limit is unknown
```

Comment:
Very easy to mistake since it requires you to read the javadoc and know that 0 is special.
I went with plain `int` in the beginning myself but after the first few test failures I changed to the OptionalInt to clarify that empty had a special meaning.